sdk/textanalytics/ai-text-analytics/samples-dev/customText.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * This sample applies actions for customer-trained models to several documents using
+ * a long-running operation. This functionality uses the generic analysis endpoint,
+ * which provides a way to group several different Text Analytics actions into a single request.
+ *
+ * @summary applies multiple Custom Text Analytics actions per document
+ * @azsdk-weight 40
+ */
+
+import {
+  TextAnalyticsClient,
+  AzureKeyCredential,
+  TextAnalyticsActions
+} from "@azure/ai-text-analytics";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// You will need to set these environment variables or edit the following values
+const endpoint = process.env["ENDPOINT"] || "<cognitive services endpoint>";
+const apiKey = process.env["TEXT_ANALYTICS_API_KEY"] || "<api key>";
+
+const documents = [
+  "Microsoft was founded by Bill Gates and Paul Allen.",
+  "Redmond is a city in King County, Washington, United States, located 15 miles east of Seattle.",
+  "I need to take my cat to the veterinarian.",
+  "The employee's SSN is 555-55-5555.",
+  "We went to Contoso Steakhouse located at midtown NYC last week for a dinner party, and we adore the spot! They provide marvelous food and they have a great menu. The chief cook happens to be the owner (I think his name is John Doe) and he is super nice, coming out of the kitchen and greeted us all. We enjoyed very much dining in the place! The Sirloin steak I ordered was tender and juicy, and the place was impeccably clean. You can even pre-order from their online menu at www.contososteakhouse.com, call 312-555-0176 or send email to order@contososteakhouse.com! The only complaint I have is the food didn't come fast enough. Overall I highly recommend it!"
+];
+
+export async function main() {
+  console.log("== Analyze Sample ==");
+
+  const client = new TextAnalyticsClient(endpoint, new AzureKeyCredential(apiKey));
+
+  const actions: TextAnalyticsActions = {
+    recognizeCustomEntitiesActions: [
+      {
+        projectName: process.env["RECOGNIZE_CUSTOM_ENTITIES_PROJECT_NAME"]!,
+        deploymentName: process.env["RECOGNIZE_CUSTOM_ENTITIES_DEPLOYMENT_NAME"]!
+      }
+    ],
+    classifyDocumentSingleCategoryActions: [
+      {
+        projectName: process.env["CLASSIFY_DOCUMENT_SINGLE_CATEGORY_PROJECT_NAME"]!,
+        deploymentName: process.env["CLASSIFY_DOCUMENT_SINGLE_CATEGORY_DEPLOYMENT_NAME"]!
+      }
+    ],
+    classifyDocumentMultiCategoriesActions: [
+      {
+        projectName: process.env["CLASSIFY_DOCUMENT_MULTIPLE_CATEGORY_PROJECT_NAME"]!,
+        deploymentName: process.env["CLASSIFY_DOCUMENT_MULTIPLE_CATEGORY_DEPLOYMENT_NAME"]!
+      }
+    ]
+  };
+  const poller = await client.beginAnalyzeActions(documents, actions, "en", {
+    includeStatistics: true
+  });
+
+  poller.onProgress(() => {
```

Comment:
In FR samples we usually show adding an initial `onProgress` handler through the options bag. Based on the rest of the sample, you must poll the operation once before the onProgress handler can be registered, so the handler won't run on the first state change this way.

Reply:
The begin operation already gives you a poller that has already been polled once. See [here](https://github.com/Azure/azure-sdk-for-js/blob/497cfdedc5d01e1c20dbeb51050af7e9ebca9aeb/sdk/textanalytics/ai-text-analytics/src/textAnalyticsClient.ts#L1111).